cmd/oras/root/attach.go

```diff
@@ -87,10 +87,19 @@ Example - Attach file to the manifest tagged 'v1' in an OCI image layout folder
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 			opts.RawReference = args[0]
 			opts.FileRefs = args[1:]
-			if err := option.Parse(cmd, &opts); err != nil {
-				return err
+			err := option.Parse(cmd, &opts)
+			if err == nil && opts.Reference == "" {
+				// ensure reference is not empty
+				err = oerrors.NewErrEmptyTagOrDigest(opts.RawReference, cmd, true)
 			}
-			return nil
+			if len(opts.FileRefs) == 0 {
+				// no file argument provided
+				if err, ok := err.(*oerrors.Error); ok && err.OperationType == oerrors.ParseArtifactReference {
+					// invalid reference
+					err.Recommendation = fmt.Sprintf("Have you specified an artifact reference to attach to? %s", err.Recommendation)
```

Comment:
Consider compiler style recommendations: "Are you missing an artifact reference to attach to"?

Reference:
- https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/assembly-references
- https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs1061

Reply:
For this case, how about improving the error message as follows: 

```
> oras attach --artifact-type test/example $ARTIFACT ./layers 
Error: "./layers" is an invalid artifact reference.
Usage: oras attach [flags] --artifact-type=<type> <name>{:<tag>|@<digest>} <file>[:<layer_media_type>] | --annotation "key=val"
To attach files to an object, please specify a valid artifact reference and files via arguments in sequence". Run "oras attach -h" for more options and examples.
```

Reply:
@FeynmanZhou What if a user is trying to attach end-of-life annotations and input an invalid artifact reference? E.g. forget to include the subject's tag. Will the error message `please specify a valid artifact reference and files via arguments in sequence` makes he/she wondering what are the `files` for?